src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/DefaultTranscriptsList.tsx

```diff
@@ -14,8 +19,54 @@ type Props = {
   rulerTicks: TicksAndScale;
 };
 
+const compareTranscriptLength = (
+  transcriptOne: Transcript,
+  transcriptTwo: Transcript
+) => {
+  const transcriptOneLength = getFeatureLength(transcriptOne);
+  const transcriptTwoLength = getFeatureLength(transcriptTwo);
+
+  if (transcriptOneLength < transcriptTwoLength) {
+    return -1;
+  }
+
+  if (transcriptOneLength > transcriptTwoLength) {
+    return 1;
+  }
+
+  return 0;
+};
+
+const sortTranscripts = (transcripts: Transcript[]) => {
+  const transcriptsWithCds = transcripts
+    .filter((transcript) => transcript.cds)
+    .sort(compareTranscriptLength);
+
+  const transcriptsWithoutCds = transcripts.filter(
+    (transcript) => !transcript.cds
+  );
+
+  const proteinCodingTranscripts = transcriptsWithoutCds
+    .filter((transcript) => transcript.biotype === 'protein_coding')
+    .sort(compareTranscriptLength);
```

Comment:
I am confused. Do such transcripts exist? Why would a transcript have a cds if it is not protein-coding?

Reply:
I am not sure and it's probably we'll need to ask during the stand-up today. I did this since it is in the criteria assuming that they can exist.